sources/osgGA/CADManipulatorStandardMouseKeyboardController.js

```diff
@@ -0,0 +1,266 @@
+'use strict';
+var OrbitManipulator = require( 'osgGA/OrbitManipulator' );
+var PolytopeIntersector = require( 'osgUtil/PolytopeIntersector' );
+var Matrix = require( 'osg/Matrix' );
+var Vec2 = require( 'osg/Vec2' );
+var ComputeMatrixFromNodePath = require( 'osg/ComputeMatrixFromNodePath' );
+
+var CADManipulatorStandardMouseKeyboardController = function ( manipulator ) {
+    this._manipulator = manipulator;
+    this._timer = false;
+    this.init();
+};
+
+CADManipulatorStandardMouseKeyboardController.prototype = {
+    init: function () {
+        this.releaseButton();
+        this._rotateKey = 65; // a
+        this._zoomKey = 83; // s
+        this._panKey = 68; // d
+        this._dimensionMask = ( 1 << 2 );
+        this._mode = undefined;
+    },
+    getMode: function () {
+        return this._mode;
+    },
+    setMode: function ( mode ) {
+        this._mode = mode;
+    },
+    setEventProxy: function ( proxy ) {
+        this._eventProxy = proxy;
+    },
+    setManipulator: function ( manipulator ) {
+        this._manipulator = manipulator;
+    },
+    setDimensionMask: function ( dimMask ) {
+        this._dimensionMask = dimMask;
+    },
+    getPositionRelativeToCanvas: function ( ev ) {
```

Comment:
Review just in case that functions can be moved to CADManipulator class, so we don't replicate code